bc_obps/service/report_service.py

```diff
@@ -50,27 +49,48 @@ class ReportService:
     @classmethod
     @transaction.atomic()
     def create_report(cls, operation_id: UUID, reporting_year: int) -> int:
-        if ReportDataAccessService.report_exists(operation_id, reporting_year):
-            raise Exception("A report already exists for this operation and year, unable to create a new one.")
-
-        # Fetching report context
-        operation = (
-            Operation.objects.select_related("operator")
-            .prefetch_related("activities", "regulated_products")
-            .get(id=operation_id)
-        )
-        operator = operation.operator
+        """
+        Creates a report and its associated report version.
+
+        If a report already exists for the given operation and reporting year, then:
+        - If there is an existing ReportVersion with a 'Draft' status, its ID is returned.
```

Comment:
Shouldn't we return an exception instead? I feel this unexpected behaviour, to return an existing object when the user expects a new report to be created.
Otherwise the endpoint would be called "fetch_or_create" (but I don't think we need to refactor this at this point)